api/client/client.go

```diff
@@ -370,6 +373,29 @@ func (c *Client) Dialer() ContextDialer {
 	return c.dialer
 }
 
+// WaitForConnectionReady waits for the client's grpc connection finish dialing,
+// returning true if it becomes ready and returning false if the ctx is canceled
+// or the connection enters a closed state. This can be used alongside the
+// DialInBackground client config option to wait until background dialing has completed.
+func (c *Client) WaitForConnectionReady(ctx context.Context) bool {
+	for {
+		switch state := c.conn.GetState(); state {
+		case connectivity.Ready:
+			return true
+		case connectivity.TransientFailure, connectivity.Connecting:
+			// The connection will begin in `TransientFailure`, then begin
+			// `Connecting` until it either succeeds as `Ready` or fails.
+			// Expect and wait for these two begginning states to change.
+			if !c.conn.WaitForStateChange(ctx, state) {
+				// ctx canceled
+				return false
```

Comment:
Log something here for debugging?

Reply:
The client doesn't have any logging and it's probably better to keep it that way. I'll change this to an error instead.